### PR TITLE
Final typing fixes + add Pyright to CI

### PR DESCRIPTION
**Pull request by tancik**

### CODE REVIEW

**Comment by brentyi** [comment]
I'll try to get the rest 🙂

**Review by brentyi — APPROVED** [review]
This is good from my end -- a skim from @jkulhanek could be helpful

File: nerfstudio/data/datamanagers/base_datamanager.py [hunk 1]
```diff
@@ -325,7 +324,7 @@ class VanillaDataManagerConfig(DataManagerConfig):
     camera_optimizer: CameraOptimizerConfig = CameraOptimizerConfig()
     """Specifies the camera pose optimizer used during training. Helpful if poses are noisy, such as for data from
     Record3D."""
-    collate_fn: Callable[[Any], Any] = staticmethod(nerfstudio_collate)
+    collate_fn: Callable[[Any], Any] = nerfstudio_collate
```
Comment (jkulhanek) [inline nerfstudio/data/datamanagers/base_datamanager.py:327]: Can you please use `field(default=nerfstudio_collate)`

File: nerfstudio/field_components/temporal_distortions.py [hunk 2]
```diff
@@ -30,7 +30,9 @@ class TemporalDistortion(nn.Module):
     """Apply spatial distortions as a function of time"""
 
     @abc.abstractmethod
-    def forward(self, positions: Float[Tensor, "*bs 3"], times: Optional[Float[Tensor, "1"]]) -> Float[Tensor, "*bs 3"]:
+    def forward(
+        self, positions: Float[Tensor, "*bs 3"], times: Optional[Float[Tensor, "*bs 1"]] = None
+    ) -> Optional[Float[Tensor, "*bs 3"]]:
```
Comment (jkulhanek) [inline nerfstudio/field_components/temporal_distortions.py:35]: I am not a fan of returning Optional here, but I guess it is ok. 
An alternative: if times is None, don't call TemporalDistortion?

**Review by jkulhanek — APPROVED** [review]

**Comment by jkulhanek** [comment]
Looks good to me. I have left some comments.

File: nerfstudio/data/datamanagers/base_datamanager.py (hunk 1 above)
Comment (brentyi) [reply nerfstudio/data/datamanagers/base_datamanager.py:327]: Do you mind explaining? Is this a static vs non-static method thing?

Comment (jkulhanek) [reply nerfstudio/data/datamanagers/base_datamanager.py:327]: Sorry, I think your code is fine.

File: nerfstudio/field_components/temporal_distortions.py (hunk 2 above)
Comment (brentyi) [reply nerfstudio/field_components/temporal_distortions.py]: Did the alternative 🙂

**Pull request merged**